### ISSUE TITLE
Support tf 2.16 and keras 3

### ISSUE THREAD

**Issue by max-pilz** [shap/shap#3672]
### Issue Description

I use Python 3.11, shap 0.45.1, and tensorflow 2.16.1. (the latter includes keras 3.3.3)
With this configuration, the example on MNIST image classification cannot be executed.
I obtain the error
`AttributeError: module 'keras._tf_keras.keras.backend' has no attribute 'learning_phase'`

### Minimal Reproducible Example

[Click here](https://shap.readthedocs.io/en/latest/example_notebooks/image_examples/image_classification/Multi-input%20Gradient%20Explainer%20MNIST%20Example.html
)

### Traceback

```shell
AttributeError: module 'keras._tf_keras.keras.backend' has no attribute 'learning_phase'
```

### Expected Behavior

The shap values should be computed as on the HTML link.

### Bug report checklist

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [ ] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [ ] I'd be interested in making a PR to fix this bug

### Installed Versions

0.45.1

**Comment by max-pilz** [comment]
Update: with tensorflow 2.15.1 the problem does not arise 
Possibly related to #3645

**Comment by CloseChoice** [comment]
Thanks for reporting, this is an important one! Will take a look

**Comment by CloseChoice** [comment]
Please note that we specifically do not support the latest tensorflow version as of now: https://github.com/shap/shap/blob/master/pyproject.toml#L74. Will use this issue as tracker to get shap ready for keras3 + tf2.16

**Comment by CloseChoice** [comment]
closed as per #3677
